sycl/source/detail/kernel_info.hpp

```diff
@@ -56,14 +56,63 @@ get_kernel_info(RT::PiKernel Kernel, const plugin &Plugin) {
 }
 
 // Device-specific methods
+template <typename Param>
+uint32_t get_kernel_device_specific_info_with_input(RT::PiKernel Kernel,
+                                                    RT::PiDevice Device,
+                                                    const void *In,
+                                                    size_t InSize,
+                                                    const plugin &Plugin) {
+  static_assert(is_kernel_device_specific_info_desc<Param>::value,
+                "Unexpected kernel_device_specific information descriptor");
+  static_assert(std::is_same<typename Param::return_type, uint32_t>::value,
+                "Unexpected return type");
+  static_assert(IsSubGroupInfo<Param>::value,
+                "Unexpected kernel_device_specific information descriptor for "
+                "query with input");
+  uint32_t Result;
+  // TODO catch an exception and put it to list of asynchronous exceptions
+  Plugin.call<PiApiKind::piKernelGetSubGroupInfo>(
+      Kernel, Device, PiInfoCode<Param>::value, InSize, In, sizeof(uint32_t),
+      &Result, nullptr);
+
+  return Result;
+}
+
+template <typename Param>
+uint32_t get_kernel_device_specific_info_with_input(RT::PiKernel Kernel,
+                                                    RT::PiDevice Device,
+                                                    sycl::range<3> In,
+                                                    const plugin &Plugin) {
+  size_t Input[3] = {In[0], In[1], In[2]};
+  return get_kernel_device_specific_info_with_input<Param>(
+      Kernel, Device, Input, 3 * sizeof(size_t), Plugin);
```

Comment:
```suggestion
  size_t Input[] = {In[0], In[1], In[2]};
  return get_kernel_device_specific_info_with_input<Param>(
      Kernel, Device, Input, sizeof(Input), Plugin);
```

IIRC what sizeof(array) does.

Reply:
The split is no longer needed and has been removed.